### PR TITLE
142 colour functions

### PR DESCRIPTION
**Pull request by jrob93**
Adding science function to estimate colours. The function `adler.science.Colour.col_obs_ref` compares a new observation in filter filt_obs to the mean of the previous N observations in filt_ref, thus estimating the filt_obs - filt_ref colour at that time.

TODO: 
- [x] Finish function demo notebook
- [ ] Integrate colour function into adler.py script
- [ ] Apply existing outlier detection methods to identify anomalous colours

N.B. there are also commits here that add extra fields to the AdlerPlanetoid class such as (heliocentricX, heliocentricY, heliocentricZ) which were being used to check ephemerides.

## Review Checklist for Source Code Changes

- [x] Does pip install still work?
- [x] Have you written a unit test for any new functions?
- [x] Do all the units tests run successfully?
- [x] Does adler run successfully on a test set of input files/databases?
- [x] Have you used black on the files you have updated to confirm python programming style guide enforcement?

### CODE REVIEW

**Comment by jrob93** [comment]
Colour demo notebook and unit test added. Note that the colour unit test will benefit from the calculated `adlerData` phase curve models being stored as part of `tests/data/testing_database.db`. I think this is related to issue #133

**Comment by jrob93** [comment]
I'm moving the integration of `col_obs_ref` into `adler.py` into a new issue #158. This is to get this PR closed and also because there is some additional reworking required to neaten up the existing implementation of `adler.py`

File: src/adler/science/Colour.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    filt_ref="r",
+    N_ref=3,
+    x_col="midPointMjdTai",
+    y_col="AbsMag",
+    yerr_col="magErr",
+    x1=None,
+    x2=None,
+):
+    """
+
```
Comment (astronomerritt) [inline src/adler/science/Colour.py:21]: This docstring doesn't describe what the function actually does: there should be a short explanation before the Parameters header.

**Review by astronomerritt — APPROVED** [review]
Approve in general, but would appreciate if you fixed the missing docstring :)

**Comment by jrob93** [comment]
Great, thanks for catching that! If the description looks okay feel free to complete the merge

**Pull request merged**